server/util-server.js

```diff
@@ -254,6 +256,32 @@ exports.mssqlQuery = function (connectionString, query) {
     });
 };
 
+/**
+ * Check domain expiry date for the specified domain
+ * @param {string} hostname Hostname / address of machine
+ * @returns {Promise<Date>} Date Object on domain expiry
+ */
+exports.whoisExpiryDate = async function (domain) {
+    const whois = await WhoisLight.lookup({ format: true }, domain);
+
+    const expiryDates = Object.entries(whois).filter(([ key, value ]) => key.match(/(Expiry|Renewal) Date$/i)).map(([ key, value ]) => value);
+
+    if (expiryDates != null) {
+        let expiry;
+        if (domain.endsWith("hk")) {
```

Comment:
Yes. 
But luckily most register return a ISO 8601 Datetime OR something can be parsed via Date.parse().
For example, we need to handle those non US datetime (eg. DD-MM-YYYY, instead of the US-preferred MM-DD-YYYY).

Hopefully a limited number of exceptions there.